connectors/protocol/connectors.py

```diff
@@ -813,7 +792,41 @@ async def prepare(self, config, sources):
         )
         await self.reload()
 
-    def updated_configuration(
+    def validated_doc(self, source_klass):
+        simple_config = source_klass.get_simple_configuration()
+        current_config = self.configuration.to_dict()
+
+        if self.configuration.is_empty():
+            # sets the defaults and the flag to NEEDS_CONFIGURATION
+            self.log_debug("Populated configuration")
+            return {
+                "configuration": simple_config,
+                "status": Status.NEEDS_CONFIGURATION.value,
+            }
+
+        missing_fields = simple_config.keys() - current_config.keys()
+        fields_missing_properties = filter_nested_dict_by_keys(
```

Comment:
I think we can avoid checking both `missing_fields` and `fields_missing_properties`:

Instead of checking
```python
for config_name, config_obj in current_config.items():
    for k, v in config_obj.items():
        simple_default_value = simple_default_config.get(config_name, {}).get(k)
```

We can check the default config first:
```python
for config_name, config_obj in simple_default_config.items():
    for k, v in config_obj.items():
        current_value = current_config.get(config_name, {}).get(k)
```

This will make sure it covers both missing fields and missing properties